src/NServiceBus.Core.Tests/Routing/MessageDrivenSubscriptions/AssemblyPublisherSourceTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System.Linq;
+    using System.Reflection;
+    using NServiceBus.Routing.MessageDrivenSubscriptions;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class AssemblyPublisherSourceTests
+    {
+        [Test]
+        public void It_returns_only_event_types()
+        {
+            var source = new AssemblyPublisherSource(Assembly.GetExecutingAssembly(), PublisherAddress.CreateFromEndpointName("Destination"));
+            var routes = source.Generate(new Conventions()).ToArray();
```

Comment:
What would happen with a user that only uses the "message" convention? Can he publish? is there a need for an opt-in like we have for the autosubscribe?